R/module_teal_with_splash.R

```diff
@@ -72,7 +80,7 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
     teal_data_rv <- if (inherits(data, "teal_data_module")) {
       data <- data$server(id = "teal_data_module")
       if (!is.reactive(data)) {
-        stop("The `teal_data_module` must return a reactive expression.", call. = FALSE)
+        stop("The `data` module must return a reactive expression.", call. = FALSE)
```

Comment:
`teal_data_module` is correct. We don't have a "data module" concept in `teal` ATM. This error message is for the app developer, who is expected to know what a `teal_data_module` is, since it is only the app developer that can pass one to `init`.

Likewise the messages below, they are to point the app developer in the right direction and need not necessarily be readily understood by the app user.

Reply:
I like better corrected suggestion but I liked current message more. No strong opinion though

Reply:
How about something very specific?
```
stop("The `teal_data_module` passed to `data` must return a reactive expression.", call. = FALSE)
```

Reply:
It's definitely more correct than the original one but I feel we shouldn't expect app developers to know what `teal_data_module` is. I would prefer to use more general "module" instead (as proposed).

Taking the dummy example from the above:  `"x argument of <abc> class needs to be of length 1"` is probably too verbose to `"x argument needs to be of length 1"`. Do you agree?

As I mentioned - it's better that the original one so I am also happy to continue with this as a compromise.

Reply:
It is impossible to get that error message without using a `teal_data_module`. Conversely, `module` - what module?

Reply:
changed as per suggestion